tests/integration/targets/filter_to_time_unit/tasks/main.yml

```diff
@@ -0,0 +1,82 @@
+---
+- name: test to_milliseconds filter
+  assert:
+    that:
+      - "('1000ms' | community.general.to_milliseconds) == 1000"
+      - "('1s' | community.general.to_milliseconds) == 1000"
+      - "('1m' | community.general.to_milliseconds) == 60000"
+
+- name: test to_seconds filter
+  assert:
+    that:
+      - "('1000msecs' | community.general.to_seconds) == 1"
+      - "('1ms' | community.general.to_seconds) == 0.001"
+      - "('12m' | community.general.to_seconds) == 720"
+      - "('300minutes' | community.general.to_seconds) == 18000"
+      - "('3h 12m' | community.general.to_seconds) == 11520"
+      - "('2days 3hours 12mins 15secs' | community.general.to_seconds) == 184335"
+      - "('2d -2d -12s' | community.general.to_seconds) == -12"
+
+- name: test to_minutes filter
+  assert:
+    that:
+      - "('30s' | community.general.to_minutes) == 0.5"
+      - "('12m' | community.general.to_minutes) == 12"
+      - "('3h 72m' | community.general.to_minutes) == 252"
+      - "('300s' | community.general.to_minutes) == 5"
+
+- name: test to_hours filter
+  assert:
+    that:
+      - "('30m' | community.general.to_hours) == 0.5"
+      - "('3h 119m 61s' | community.general.to_hours) > 5"
+
+- name: test to_days filter
+  assert:
+    that:
+      - "('1year' | community.general.to_days) == 360"
+      - "('1week' | community.general.to_days) == 7"
+      - "('2weeks' | community.general.to_days) == 14"
+      - "('1mo' | community.general.to_days) == 30"
+
+- name: test to_weeks filter
+  assert:
+    that:
+      - "('1y' | community.general.to_weeks | round(0, 'ceil')) == 52"
+      - "('7d' | community.general.to_weeks) == 1"
+
+- name: test to_months filter
+  assert:
+    that:
+      - "('30d' | community.general.to_months) == 1"
+      - "('1year' | community.general.to_months) == 12"
+
+- name: test to_years filter
+  assert:
+    that:
+      - "('360d' | community.general.to_years) == 1"
+      - "('12mo' | community.general.to_years) == 1"
```

Comment:
```suggestion
      - "('12mo' | community.general.to_years) == 1"
      - "('11mo' | community.general.to_years) == ?"
      - "('993849days' | community.general.to_years) == ?"
```

Reply:
just for better coverage:)

Reply:
I'm wondering whether it is a good idea to have month and year support. The problem is that timespans of years and months only make sense when talking about a concrete starting (or ending) point of the interval. If you ignore this and use something like `a month == 30 days` or `a month == 30.5 days` or `a month == 30.41666... days` or `a month == 30.4375` days: all of these choices make sense - but only for some use-cases, and not for others. The same for years. You can define them as `12 * number of days per month` days, as `365 days`, as `365.25 days`, etc. Every choice makes sense for some use-case, but not for others.

The usual solution for this problem is to not support months and years for types representing timespans (see [Python's timedelta](https://docs.python.org/3/library/datetime.html#timedelta-objects), [Java's Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html), ...). That's why it may be best to not support it here either, or only when the user provides further information (for example `month=30, year=365` as keyword args, and the function will complain if you use month or year as a unit and the corresponding keyword arg is not there).

Reply:
so my thinking was not to calculate the exact time for a year or a month or a day (a day has ~ 23 hours and 56 minutes btw).
I want to provide a human normalized form of a time:

I learned in german accounting 20 years ago (and accounting uses time ranges in the most human normalized form I know) a year has 12 months, each month has 30 days. a week has 7 days and a day has 24 hours. And why this? because it makes calculation easy.

However, not every country uses 360 days for a banking year. So I would like to keep 1mon=30days but would be fine to say 365 days = 1 year instead of 360.

The poll is opened ;)

Reply:
That's indeed one valid way to interpret what a month or year is. But this is not what every user expects. I have no idea how many would expect what, and no matter which concrete value you chose, probably sooner or later someone will complain that your choice is "wrong" ;-)

That's why my suggestion would be to either drop support for months and years, or make it configurable. (I would also be happy if there are default values, like `month=30` and `year=360`, as long as you can override them if you want with values of your choice.)

Reply:
sounds reasonable to me

Reply:
someone will complain is not an arg ;) whatever default we use, I am sure, someone would complain we choose the wrong default. 

Anyways. this pr took way more time than I had, My suggestion is to merge it as is and you work on top of it (or on top of this PR).

I won't invest more time.

Reply:
Sounds like a plan! :)

Reply:
i'll make a note on a piece of paper :)